i3bar/src/xcb.c

```diff
@@ -184,46 +185,55 @@ static void draw_separator(i3_output *output, uint32_t x, struct status_block *b
+static uint32_t predict_block_length(struct status_block *block) {
+    i3String *text = block->full_text;
+    struct status_block_render_desc *render = &block->full_render;
+    if (block->use_short && block->short_text != NULL) {
+        text = block->short_text;
+        render = &block->short_render;
+    }
 
-    TAILQ_FOREACH (block, &statusline_head, blocks) {
-        i3String *text = block->full_text;
-        struct status_block_render_desc *render = &block->full_render;
-        if (use_short_text && block->short_text != NULL) {
-            text = block->short_text;
-            render = &block->short_render;
+    if (i3string_get_num_bytes(text) == 0)
+        return 0;
+
+    render->width = predict_text_width(text);
+    if (block->border)
+        render->width += logical_px(block->border_left + block->border_right);
+
+    /* Compute offset and append for text alignment in min_width. */
+    if (block->min_width <= render->width) {
+        render->x_offset = 0;
+        render->x_append = 0;
+    } else {
+        uint32_t padding_width = block->min_width - render->width;
+        switch (block->align) {
+            case ALIGN_LEFT:
+                render->x_append = padding_width;
+                break;
+            case ALIGN_RIGHT:
+                render->x_offset = padding_width;
+                break;
+            case ALIGN_CENTER:
+                render->x_offset = padding_width / 2;
+                render->x_append = padding_width / 2 + padding_width % 2;
+                break;
         }
+    }
 
-        if (i3string_get_num_bytes(text) == 0)
-            continue;
+    block->render_length = render->width + render->x_offset + render->x_append;
+    return block->render_length;
+}
 
-        render->width = predict_text_width(text);
-        if (block->border)
-            render->width += logical_px(block->border_left + block->border_right);
+static uint32_t predict_statusline_length(void) {
+    uint32_t width = 0;
+    struct status_block *block;
 
-        /* Compute offset and append for text alignment in min_width. */
-        if (block->min_width <= render->width) {
-            render->x_offset = 0;
-            render->x_append = 0;
-        } else {
-            uint32_t padding_width = block->min_width - render->width;
-            switch (block->align) {
-                case ALIGN_LEFT:
-                    render->x_append = padding_width;
-                    break;
-                case ALIGN_RIGHT:
-                    render->x_offset = padding_width;
-                    break;
-                case ALIGN_CENTER:
-                    render->x_offset = padding_width / 2;
-                    render->x_append = padding_width / 2 + padding_width % 2;
-                    break;
-            }
-        }
+    TAILQ_FOREACH (block, &statusline_head, blocks) {
+        uint32_t block_width = predict_block_length(block);
+        if (block_width == 0)
+            continue;
```

Comment:
nit: use curly brackets
```suggestion
        if (block_width == 0) {
            continue;
        }
```

Reply:
Updated in 65e9333644a382debe2a5a52802399785f1c5b87. This doesn't seem to be consistently enforced, though. Perhaps the `.clang-format` rules could be updated to require use of braces even for single line blocks, e.g. with [`InsertBraces`](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#insertbraces).

Reply:
Addressed in #5882